code/ZenValidatorConstraint.php

```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
+                                break;
+                            case self::MAX_WIDTH_HEIGHT:
+                                if (($width <= $this->val1) && ($height <= $this->val2)) {
+                                    return true;
+                                } else {
+                                    return false;
+                                }
+                                break;
+                            default:
+                                throw new Exception('Invalid type : ' . $this->type);
+                        }
+                    }
+                }
+            }
+        }
+
+        exit();
+    }
+
+    /**
+     * Does the ratio validation
+     * @param  Int $width  width of the image
+     * @param  Int $height height of the image
+     * @return boolean
+     */
+    protected function ratioValidation($width,$height)
+    {
+        $validationResult = false;
+
+        // Divide the width and height by the ratio, ignore the decimal points as can't have part pixles.
+        $baseWidth = floor($width / $this->val1);
+        $baseHeight = floor($height / $this->val2);
+
+        // Compare the base values, if they match then the image is in the correct aspect ratio.
+        if ($baseWidth == $baseHeight) {
+            $validationResult = true;
+        }
+
+        return $validationResult;
+    }
+
+    /**
+     * Gets the default message for the validator
+     * @return String the validation message
+     */
+    public function getDefaultMessage()
+    {
+        switch ($this->type) {
+            case self::WIDTH:
+                return sprintf(_t('ZenValidator.DIMWIDTH', 'Image width must be %s pixles'), $this->val1);
+                break;
+            case self::HEIGHT:
+                return sprintf(_t('ZenValidator.DIMHEIGHT', 'Image height must be %s pixles'), $this->val1);
+                break;
+            case self::WIDTH_HEIGHT:
+                return sprintf(_t('ZenValidator.DIMWIDTHHEIGHT', 'Image width must be %s pixles and Image height must be %s pixles'), $this->val1, $this->val2);
+                break;
+            case self::RATIO:
+                return sprintf(_t('ZenValidator.DIMRATIO', 'Image aspect ratio (shape) must be %s:%s'), $this->val1, $this->val2);
+                break;
+            case self::MIN_WIDTH:
+                return sprintf(_t('ZenValidator.DIMMINWIDTH', 'Image width must be greater than or equal to %s pixles'), $this->val1);
+                break;
+            case self::MIN_HEIGHT:
+                return sprintf(_t('ZenValidator.DIMMINHEIGHT', 'Image height must be greater than or equal to %s pixles'), $this->val1);
+                break;
+            case self::MIN_WIDTH_HEIGHT:
+                return sprintf(_t('ZenValidator.DIMMINWIDTHHEIGHT', 'Image width must be greater than or equal to %s pixles and Image height must be greater than or equal to %s pixles'), $this->val1, $this->val2);
```

Comment:
some of the lines look pretty long, can we bring it under 100 chars? just use multiple lines...